### PR TITLE
New component: sb-editor-header

### PR DESCRIPTION
**Pull request by ademarCardoso**

### CODE REVIEW

File: src/components/EditorHeader/EditorHeader.vue [hunk 1]
```diff
@@ -0,0 +1,218 @@
+<template>
+  <div class="sb-editor__container">
+    <SbHeaderTitle :title="headerTitle" :format="returnViewFormat" />
+
+    <div
+      class="sb-editor-header"
+      :class="{ 'sb-editor-header__mobile': returnViewFormat === 'mobile' }"
+    >
+      <SbDesktopAndTabletViwer
```
Comment (gustavomelki) [inline src/components/EditorHeader/EditorHeader.vue:9]: I think the correct is SbDesktopAndTablet**Viewer**

File: src/components/EditorHeader/EditorHeader.vue [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+          headerSubTitle,
+          languages,
+          users,
+          actions,
+          format: returnViewFormat,
+          size: window.width,
+        }"
+        @changes="$_handleListenersOrchestrator"
+      />
+      <SbMobileViwer
```
Comment (gustavomelki) [inline src/components/EditorHeader/EditorHeader.vue:22]: same here: SbMobile**Viewer**

File: src/components/EditorHeader/EditorHeader.vue [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      return this.spaceStatus === 'unpublished'
+        ? 'light'
+        : this.spaceStatus === 'published'
+        ? 'primary'
+        : this.spaceStatus === 'published-and-changes'
+        ? 'light-gray'
+        : 'light'
+    },
+
+    returnViewFormat() {
```
Comment (gustavomelki) [inline src/components/EditorHeader/EditorHeader.vue:144]: Maybe we need to create something more elaborate, or use some plugin to define breakpoints. But let's leave it at that for now.

File: src/components/EditorHeader/EditorHeader.vue [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  created() {
+    window.addEventListener('resize', this.$_handleResize)
+    this.$_handleResize()
+  },
+
+  destroyed() {
+    window.removeEventListener('resize', this.$_handleResize)
+  },
+
+  methods: {
```
Comment (gustavomelki) [inline src/components/EditorHeader/EditorHeader.vue:162]: jump lines between methods

File: src/components/EditorHeader/EditorHeader.vue [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+     */
+    handleSetNewLanguage(lang) {
+      this.$emit('change-language', { language: lang })
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+@import './editor-header.scss';
```
Comment (gustavomelki) [inline src/components/EditorHeader/EditorHeader.vue:217]: Are we importing the styles inside the component?

File: src/components/EditorHeader/__tests__/DesktopAndTabletViwer.spec.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  size: 1200,
+  headerTitle: 'A awesome cms',
+  headerSubTitle: 'Its true',
+}
+
+describe('Test if SbDesktopAndTabletViwer renderer correctly', () => {
+  it('Test if desktop and tablet view render all components correctly', () => {
+    const wrapper = factory(fakeProps)
+
+    expect(wrapper.findComponent(SbHeaderItem).exists()).toBe(true)
```
Comment (gustavomelki) [inline src/components/EditorHeader/__tests__/DesktopAndTabletViwer.spec.js:54]: How about comment the expected behavior in tests?

File: src/components/EditorHeader/__tests__/EditorHeader.spec.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+import SbIcon from '../../Icon'
+import SbButton from '../../Button'
+
+import {
+  SbMenu,
+  SbMenuButton,
+  SbMenuList,
+  SbMenuItem,
+  // SbMenuGroup,
```
Comment (gustavomelki) [inline src/components/EditorHeader/__tests__/EditorHeader.spec.js:17]: remove?

File: src/components/EditorHeader/__tests__/EditorHeader.spec.js [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  actions: [...actionsList],
+  options: [...optionsList],
+  spaceStatus: 'unpublished',
+  hasSaveButton: true,
+  showPublishedIcon: true,
+  headerTitle: 'A awesome CMS',
+  headerSubTitle: 'Its true',
+}
+
+describe('Test if SbEditorHeader renderer correctly', () => {
```
Comment (gustavomelki) [inline src/components/EditorHeader/__tests__/EditorHeader.spec.js:84]: same here: How about comment the expected behavior in tests?

File: src/components/EditorHeader/components/DesktopAndTabletViwer.vue [hunk 9]
```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="sb-editor-header__desktop">
+    <div v-if="format === 'desktop'" class="header__desktop">
+      <span class="desktop__title">{{ headerTitle }}</span>
+      <span class="desktop__sub-title">{{ headerSubTitle }}</span>
+    </div>
+    <SbHeaderItem v-if="users" style="min-width: 70px">
```
Comment (gustavomelki) [inline src/components/EditorHeader/components/DesktopAndTabletViwer.vue:7]: In components, I think is better define styles inside files, not inline.

File: src/components/EditorHeader/components/DesktopAndTabletViwer.vue [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+            </SbMenuItem>
+          </SbMenuGroup>
+        </SbMenuList>
+      </SbMenu>
+    </SbHeaderItem>
+
+    <div class="sb-editor-header__actions">
+      <SbHeaderItem
+        v-for="act in actions"
+        v-show="size > 765"
```
Comment (gustavomelki) [inline src/components/EditorHeader/components/DesktopAndTabletViwer.vue:38]: this value must be inside data or computed, not hardcoded.

File: src/components/EditorHeader/components/DesktopAndTabletViwer.vue [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+        :key="act.id"
+        with-separator
+      >
+        <button class="action__button" @click="handleSelectNewAction(act.name)">
+          <SbTooltip position="bottom" :label="act.name">
+            <SbIcon :name="act.name.toLowerCase()" size="large" />
+          </SbTooltip>
+        </button>
+      </SbHeaderItem>
+      <SbHeaderItem v-if="size < 765" with-separator>
```
Comment (gustavomelki) [inline src/components/EditorHeader/components/DesktopAndTabletViwer.vue:48]: same here

File: src/components/EditorHeader/components/DesktopAndTabletViwer.vue [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+        <SbMenu>
+          <SbMenuButton
+            v-if="size < 700"
```
Comment (gustavomelki) [inline src/components/EditorHeader/components/DesktopAndTabletViwer.vue:51]: different value for breakpoint?

**Review by gustavomelki — CHANGES_REQUESTED** [review]
There are a lot of different values defining the breakpoints. It is necessary to standardize.

File: src/components/EditorHeader/EditorHeader.vue (hunk 5 above)
Comment (ademarCardoso) [reply src/components/EditorHeader/EditorHeader.vue:217]: My mistake, I forgot to remove

**Review by gustavomelki — APPROVED** [review]
Nice work! 
One point: how about create the variations isOnTabletOrDesktop, isOnMobileOrTablet, etc directly in the mixin file? Because these variations will be used a lot and we don't need to be creating computeds in each component.

**Review by emanuelgsouza — APPROVED** [review]

**Pull request merged**